TLM/TLM/Manager/Impl/RoutingManager.cs

```diff
@@ -2342,5 +2346,45 @@ bool laneChangesAllowed
 
             RecalculateAll();
         }
+
+        /// <summary>
+        /// if the direction of the given lanes do not match then similar lane index must be reversed.
+        /// </summary>
+        private bool ShouldReverseSimilarLaneIndex(
+            bool segmentInvert1, NetInfo.Lane laneInfo1, bool startNode1,
+            bool segmentInvert2, NetInfo.Lane laneInfo2, bool startNode2) {
+#if DEBUG
+            bool logRouting = DebugSwitch.Routing.Get(); 
+#else
+            const bool logRouting = false;
+#endif
+            bool both1 = laneInfo1.m_finalDirection == NetInfo.Direction.Both;
+            bool both2 = laneInfo2.m_finalDirection == NetInfo.Direction.Both;
+            bool backward1 = laneInfo1.IsGoingBackward();
+            bool backward2 = laneInfo2.IsGoingBackward();
+
+            bool reverse;
+            if (!both1 && !both2) {
+                // both lanes are one-way or station tracks
+                // [https://github.com/CitiesSkylinesMods/TMPE/issues/1486#issuecomment-1075699771] what about station tracks connecting to unidirectional tracks?
+                reverse = false;
+            } else {
+                // at least one lane is non-station bidirectional
+                reverse = startNode1 == startNode2; // if the common node is start node for both segments then they are facing each other and are in opposite direction.
+                reverse ^= segmentInvert1 != segmentInvert2;
+                reverse ^= backward1 != backward2;
```

Comment:
Why not the boring if else? This part is almost as unreadable as it was possible to be written.

| P | Q |  `XOR` |
|---|---|---|
|1|1|`0`|
|1|0|`1`|
|0|1|`1`|
|0|0|`0`|

Reply:
I prefer the logic operators tbh, much more readable.

Reply:
I prefer the logic operators tbh, much more readable.

Reply:
@krzychu124 why is it unreadable?
1-if only `startNode1 == startNode2` => segments are facing each other => need to reverse lane index
2-if only `segmentInvert1 != segmentInvert` => segments are in different directions => need to reverse lane index
3-if only `backward1 != backward2` => lanes are in different directions => need to reverse lane index

if you change direction two times it is as if direction did not changed. so that is why we need xor. its like toggle.
I can put more comments.

> Why not the boring if else?

3 conditions => 9 different branches. not only that it would be slow but also it would not make much sense and it would be unreadable.

Why did you include the truth table for xor? You are thinking too much! You should think of `^=` as toggle operator:
```cs
x ^= true => x = ~x
x ^= false => x = x
```

PS: I admit that this part of the logic is hard to understand. But the again, direction logic is typically supper confusing in CS. luckily I had a lot of experience with this direction madness!

Reply:
Just add some code comment? See my suggested change below.

Reply:
is it better now?